submissions/mfialko/js-oop/js/app.js

```diff
@@ -0,0 +1,158 @@
+
+const //borders for enemy respawn
+    enemyLeftX = -100,
+    enemyRightX = 500,
+    //size of step
+    stepX = 101,
+    stepY = 83,
+    //borders for player
+    playerLeftX = 0,
+    playerRightX = 400,
+    playerTopY = 0,
+    playerBottomY = 380,
+    //player starting position
+    playerPosX = 202,
+    playerPosY = 380,
+    //deltas
+    deltaForEnemy = 60, 
+    deltaForGem = 15;
+
+
+const Enemy = function(row, speed) {
+    this.x = enemyLeftX;
+    this.y = deltaForEnemy + (row-1)*stepY;
+    this.speed = speed;
+    // The image/sprite for our enemies
+    this.sprite = 'images/enemy-bug.png';
+};
+
+
+// Update the enemy's position
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    this.x = this.x + this.speed * dt;
+    if (this.x > enemyRightX) {this.x = enemyLeftX;
+    }
+};
+
+// Draw the enemy on the screen
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Player = function() {
+    this.x = playerPosX;
+    this.y = playerPosY;
+    this.score = 0;
+    this.highScore = 0;
+    this.sprite = 'images/char-boy.png';
+
+};
+
+const Gem = function() {
+    this.x = stepX * getRandNum(0,5)+deltaForGem;
+    this.y = stepY * getRandNum(1,4);
+    this.sprite = 'images/GemOrange.png';
+}
+Gem.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+}
+Gem.prototype.get = function(score = 5) {
```

Comment:
Avoid using "magic numbers". I guess it would be better like `const bonusToScore` or smth like that.

Reply:
Added some const variables for this and also several comments to explain purpose of numbers.